jib-core/src/main/java/com/google/cloud/tools/jib/http/Authorization.java

```diff
@@ -34,9 +46,66 @@
    * @return an {@link Authorization} with a {@code Bearer} token
    */
   public static Authorization fromBearerToken(String token) {
-    return new Authorization("Bearer", token);
+    return new Authorization("Bearer", token, decodeTokenRepositoryGrants(token));
   }
 
+  /**
+   * Decode the <a href="https://docs.docker.com/registry/spec/auth/jwt/">Docker Registry v2 Bearer
```

Comment:
Let's not interleave `decodeTokenRepositoryGrants` and `isValidAccessClaim` between the `fromXXX` static initializer. Put it after the last `fromXXX`. Maybe moving `fromBearerToken` to make it the last will help.